src/helpers/Helpers.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/**
+ * Commerce Insights Components Helpers class
+ *
+ * @link      https://fostercommerce.com
+ * @copyright Copyright (c) 2021 Foster Commerce
+ */
+
+declare(strict_types = 1);
+
+namespace fostercommerce\commerceinsights\helpers;
+
+use DateTime;
+use NumberFormatter;
+use Money\Money;
+use Money\Currency;
+use Money\Currencies\ISOCurrencies;
+use Money\Formatter\IntlMoneyFormatter;
+
+use Craft;
+
+class Helpers {
+    public static function getDateRangeData(): array {
+        $result = [];
+
+        // Query params if they exist in the URL
+        $startQuery = Craft::$app->request->getQueryParam('startDate') ?? null;
+        $endQuery   = Craft::$app->request->getQueryParam('endDate') ?? null;
```

Comment:
Will it get called from a console action? Why or why not? The problem is that all of the logic for this method would essentially be useless because the main point of it is to prevent having to get the posted dates and/or query param dates all over the app (and also it keeps things consistent because we always need the query params, if present, to override anything in `POST`, or maybe it's vice versa, I forget off-hand, but we want it to always be consistent application-wide in how it deals with determining which ones are used when multiple possibilities are present).